File: Back/quiz-app/src/main/java/com/quiz/quiz_app/Entity/Activity.java
package com.quiz.quiz_app.Entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "atividade")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Activity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @Column(name = "titulo", nullable = false)
    private String title;

    @Column(name = "descricao")
    private String description;

    @Column(name = "ano")
    private Integer activityYear;

    @ManyToOne
    @JoinColumn(name = "criado_por", nullable = true)
    private User createdBy;

    @Column(name = "created_at", updatable = false, nullable = false)
    private LocalDateTime createdAt;

    @PrePersist
    public void prePersist() {
        this.createdAt = LocalDateTime.now();
    }
}

File: Front/src/interfaces/Activity.ts
export interface ActivityResponseDTO {
  id: number;
  title: string;
  description: string;
  createdBy: number;
  createdAt: Date;
  activityYear: number;
}

export interface ActivityRequestDTO {
  title: string;
  description: string;
  createdById: number;
  activityYear: number;
}

export interface ActivityAnswered {
  id: number;
  title: string;
  description: string;
  status: string | null;
}